lib/passport-configurator.js

```diff
@@ -300,15 +300,27 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
                 }
               } else {
                 user.hasPassword(password, function(err, ok) {
-                  if (ok) {
-                    done(null, userProfile);
+                  if (ok) {          
+                     if (!user.emailVerified) 
+                      {
+                          // Fail to log in if email verification is not done yet
+                          return done(null, false, {message: 'User Email has not been verified.'});
+                          err = new Error('login failed as the email has not been verified');
```

Comment:
I tested using `loopback-example-passport` and test using this branch of loopback-component-passport`
it gave the correct behavior for verified and non-verified email. Where's the mistake?